src/main/java/me/jellysquid/mods/lithium/common/util/TypeCast.java

```diff
@@ -0,0 +1,17 @@
+package me.jellysquid.mods.lithium.common.util;
+
+import net.minecraft.world.explosion.Explosion;
+
+/**
+ * Class to work around being unable to cast SomeClassMixin to SomeClass inside SomeClassMixin.
+ * Add methods for ohter types when needed.
+ *
+ * Hopefully(!) the casts can be inlined and completely eliminated (casting SomeClass to SomeClass never doing anything).
+ *
+ * @author 2No2Name
+ */
+public class TypeCast {
+    public static Explosion toExplosion(Object returnValue) {
```

Comment:
Why not just inline this cast?

Reply:
I usually get problems when trying to cast to the mixined class. Either intellij, mcdev or the compiler complains, or it crashes when applying the mixin

Reply:
inconvertible types, because ExplosionMixin doesn't extend Explosion.
This can be circumvented by casting like this: (Explosion)(Object)this
But I had problems with that approach in the past, but I don't recall when exactly they occur.

Reply:
Why is it cast to the Mixin in the first place? I thought you should cast it to an interface which the mixin implements?

Reply:
Even then, I would have made it a private function in the mixin, instead of making a static class just for this.